### PR TITLE
[DEC-1209] Add fork workflow instructions

### PR DESCRIPTION
**Pull request by prettymuchbryce**
Proposed workflow instructions for managing forks.

* Removes the need for manually tagging every commit with increasingly obtuse tag names.
* Showcases our current fork as the default branch for readability. See [here](https://github.com/dydxprotocol/tendermint).
* Adds instructions around the process for updating, and using GitHub's "Sync Fork" feature.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
+# dYdX Fork of Tendermint
+
+This is a lightweight fork of Tendermint. Forked code resides on the default branch.
```
Comment (ttl33) [inline README.md]: Can we clarify what the default branch means here? Does it refer to the latest forked tendermint branch that we are using?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+## Fork maintenance
+
+We'd like to keep the `master` branch up to date with `tendermint/tendermint`. You can utilize GitHub's [sync fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) button to accomplish this. ⚠️ Please only use this on the `master` branch, not on the fork branches as it will discard our commits.⚠️
+
+## Updating Tendermint to new versions
+
+When a new version of Tendermint is published, we may want to adopt the changes in our fork. This process can be somewhat tedious, but below are the recommended steps to accomplish this.
+
+1. Ensure the `master` branch is up to date by following the steps above in "Fork maintenance".
+2. Create a new branch off the desired `master` commit. It should be named something like `dydx-fork-$VERSION` where `$VERSION` is the version of Tendermint being forked. i.e. `dydx-fork-v0.37.0-rc2`.
```
Comment (ttl33) [inline README.md]: Do we want to create a branch off of the latest Tendermint `master`? They typically release new versions using tags (i.e. `v0.37.0-rc2`) and `master` branch can contain commits that are not part of `v0.37.0-rc2` yet and will be part of the next version. Also, I noticed that they often cherry-pick some changes onto these tags. 

IMO, I think we should create our fork based off of their tags.  

Screen Shot 2023-01-31 at 9 33 02 AM

Comment (ttl33) [reply README.md]: I typically do the following:

```
# Update forked repo w original repo's tags
git fetch upstream
git push --tags

# Create a new branch off of an existing tag
git checkout -b <branch name> <origina repo's tag name>
(i.e. git checkout -b dydx-fork-v0.37.0 v0.37.0)
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+3. Open a PR which cherry-picks each commit in the current default branch, in order, on to `dydx-fork-$VERSION` (note: you may want to consider creating multiple PRs for this process if there are difficulties or merge conflicts).
```
Comment (ttl33) [inline README.md]:
```suggestion
3. Open a PR which cherry-picks each commit in the current default branch, in order, on to `dydx-fork-$VERSION` (note: you may want to consider creating multiple PRs for this process if there are difficulties or merge conflicts). For example, `git cherry-pick <commit hash>`
```

**Review by ttl33 — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (prettymuchbryce) [reply README.md]: Default branch is a concept of GitHub outlined [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#about-the-default-branch).

It's the branch that is displayed by default when you visit the repository. I can clarify further.

Screen Shot 2023-01-31 at 3 08 36 PM

**Pull request merged**